### ISSUE TITLE
Missing letters when using type 

### ISSUE THREAD

**Issue by Mattinn** [cypress-io/cypress#3817]
### Current behavior:
I just upgraded to Cypress 3.2.0 and noticed this strange behavior with type    

![missingFirstLetter](https://user-images.githubusercontent.com/2447396/54931168-300a3080-4f21-11e9-9aea-412c2fe32b61.gif)

For some reason the first letter gets cut out, this test used to run fine in 3.1.5.  

### Desired behavior:
That the typed text does not get cut short

### Steps to reproduce: (app code and test code)
The test
```javascript
  it('Successful login', () => {
    cy.visit('/')
    cy.get('head title')
    .should('contain', 'Log in')
    cy.get('#email')
      .type('Admin{enter}', {force: true, delay: 700})
    cy.get('#password')
      .type('superSafePassword{enter}')
    cy.get('.primary').click()
    cy.get('#title')
      .should('contain', 'Dashboard')
  });
});
```

The form 
```javascript
<div class='right'>
  <p class='greeting'>Log in to your account</p>
  <ListErrors {errors}/>

  <form on:submit='submit(event)' autocomplete='off'>
    <fieldset class='login'>
      <img alt='user' src='img/user.svg' class='login-icon' />
      <input type='text' placeholder='Email' id='email' autocomplete='off' bind:value=email>
    </fieldset>

    <fieldset class='login'>
      <img alt='password' src='img/password.svg' class='login-icon' />
      <input type='password' placeholder='Password' id='password' autocomplete='new-password' bind:value=password>
    </fieldset>
    <Button kind='primary block' type='submit' disabled='{!email || !password}'>Log in</Button>
  </form>
</div>
```

### Versions
Windows 10
Cypress 3.2.0
Chrome 73.0.3683.86 (Official Build) (64-bit)

**Comment by Mattinn** [comment]
I added `cy.wait(500);` in the beginning of the test and now it passes. Seems like Cy runs just a bit too fast :)  What is the recommended action when dealing with issues like this? I dont really want to add wait to all of my test cases

**Comment by bahmutov** [comment]
@Mattinn this is a common question and a problem. See my two blog posts about it
- https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/
- https://www.cypress.io/blog/2018/02/05/when-can-the-test-start/

**Comment by Mattinn** [comment]
Thanks Gleb, these posts explain the problem very well 👍

**Comment by AngelRajan** [comment]
<img width="581" alt="Screenshot 2020-06-10 at 1 23 10 PM" src="https://user-images.githubusercontent.com/34320843/84217272-cecb7e80-ab1f-11ea-965d-d9beb51966a2.png">
I'm still having same issue 
I try to 
type('AID')
type('First name') etc and the first letters are skipping

**Comment by davidyell** [comment]
I found this issue when Googling for why my field is missing the first few characters of my type.

```javascript
cy.get('#customer-postcode').type('W1A 1AA');
cy.get('#customer-postcode').should('have.value', 'W1A 1AA');
```

I expected to see `W1A 1AA` but usually all I get is `A 1AA` or a variation thereof. I have had to add a check for the value to force the test to fail when Cypress cannot correctly fill the field.

**Comment by t0lkman** [comment]
same here

**Comment by kpassoubady** [comment]
This is not flaky cypress 4.11.0 has the same issue

**Comment by jennifer-shehane** [comment]
Some of you may be experiencing this issue: https://github.com/cypress-io/cypress/issues/5480

This specific issue that the original poster was experiencing has been closed as fixed however. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.

**Comment by davidyell** [comment]
I solved my issue by adding `cy.wait(500)` before trying to type. Presumably a vagary of my interface.

**Comment by AngelRajan** [comment]
> I solved my issue by adding `cy.wait(500)` before trying to type. Presumably a vagary of my interface.

I don't think it is a good idea to keep hardcoded wait before every type.

**Comment by t0lkman** [comment]
> > I solved my issue by adding `cy.wait(500)` before trying to type. Presumably a vagary of my interface.
> 
> I don't think it is a good idea to keep hardcoded wait before every type.

so what the alternative ?

**Comment by kpassoubady** [comment]
By using a work-around like cy.wait() some time the above issue disappears, but the original issue remains the same.

**Comment by AngelRajan** [comment]
> By using a work-around like cy.wait() some time the above issue disappears, but the original issue remains the same.

Yeah. Exactly.

**Comment by kpassoubady** [comment]
Here is another workaround which worked for me.
I have used tab along with focus and clear to move between input fields.
```javascript:
      cy.get('div[class*=first_name] input').type('Kangs').should('have.value', 'Kangs').tab()
      cy.get('div[class*=last_name] input').focus().clear().type('Passoubady').should('have.value', 'Passoubady').tab()
```
look for cypress-plugin-tab [here](https://github.com/Bkucera/cypress-plugin-tab)

**Comment by ashwinisambandan** [comment]
I had an issue with password not matching. Here's what i tried 

`cy.get("#password").find("[type='password']").wait(500).clear().type('password',{delay: 700})
 cy.get("#repeat-password").find("[type='password']").type('password')`

**Comment by AngelRajan** [comment]
> I had an issue with password not matching. Here's what i tried
> 
> `cy.get("#password").find("[type='password']").wait(500).clear().type('password',{delay: 700}) cy.get("#repeat-password").find("[type='password']").type('password')`
Does cy.get("#password").wait(500).type('password') work fine?

**Comment by bahmutov** [comment]
Hey, I see people commenting on this issue, despite it being closed. If you have a reproducible example you can share with us (so we can run it ourselves to see the problem), please open a new issue.

**Comment by shirshendu30** [comment]
> Hey, I see people commenting on this issue, despite it being closed. If you have a reproducible example you can share with us (so we can run it ourselves to see the problem), please open a new issue.

If you try to type a big text and use back to back type for 4-5 different fields you might see the issue. using Cypress 6.4.0.

**Comment by jennifer-shehane** [comment]
@shirshendu30 Please provide code - HTML, CSS, and JS and the Cypress tests that fail with it. Open a new issue.

**Comment by shirshendu30** [comment]
describe('Check Insight Present in Filtertab', function () {
    const explorerPage = new Explorer();
    const login = new Login();
    let filterOptions: string[] = [];
    let check: boolean = false;

    it('Explorer Insight verification', () => {

        cy.visit('url');
        login.email().type('longUserName- 50 chrs');
        login.password().type('longPwd - 40chars');
        login.signInButton().should('be.visible').click();
        explorerPage.filterbutton().should('be.visible').click();
    })

    it('verify Product, Folder, Test in source Explorer', () => {
        
        cy.visit('url');
        login.email().type('longUserName- 50 chrs');
        login.password().type('longPwd - 40chars');
        login.signInButton().should('be.visible').click();
        cy.visit('someotherurl');

    })
})


Mostly when we have two 'it' , 2nd test('it') is only wrting first few characters of the username and password. This is intermittent. mostly seem when used chrome browser.

**Comment by jennifer-shehane** [comment]
@shirshendu30 We cannot run the code you've provided - there is no HTML/CSS or site to visit. We cannot look into the issue without being able to reproduce it ourselves.

**Comment by jonnyparris** [comment]
I'm hitting this bug for the 4th time in 12 months of various fresh installations.
1) the WORKAROUND (_without_ using an arbitrary `wait()`) that I use is simply to duplicate the very first input `type()` command. Something like the following:
``` javascript
    cy.get("input[name=email]").type("x"); // dummy input to "warm up" Cypress
    cy.get("input[name=email]").clear().type(email);
    cy.get("input[name=password]").type(password);
    cy.get("button[type=submit]").click();
```
2) This is has been so consistent it's hard to believe that no maintainers can reproduce it; but here's some markup (pug) to run it against that hopefully can help debug (@jennifer-shehane 🙏🏾 ). It's essentially just an input for email and password with a button to submit the form
``` pug
form
    div
      label(for="email") Email
      input(
        name="email",
        placeholder="Email",
        type="email",
        autocomplete="username"
      )
    div
      label(for="password") Password
      input(
        name="password",
        placeholder="Password",
        type="password",
        autocomplete="current-password"
      )
    button(type="submit") Log in
```

Really hope this can help because persistent bugs like this for what should be simple example really trip me up when trying to evangelise about the joys & ease of testing with awesome tools like Cypress.

Hope this can help any other googlers or at least help me converge towards this workaround much quicker next time 😅

**Comment by akwasin** [comment]
This is still an issue. 

`cy.get('#ContentPlaceHolderMain_Wizard1_SelectPatient_TextBoxSocialNumber').type('19760824-8683').type('{enter}')`

<img width="758" alt="Screenshot 2021-03-18 at 15 35 32" src="https://user-images.githubusercontent.com/13676596/111643910-a041a900-87ff-11eb-81fc-46183f3fe23b.png">

I can PM link to test site if needed.

**Comment by jennifer-shehane** [comment]
@akwasin If you have a reproducible example that we can run completely and see the error, you can email it so support@cypress.io

**Comment by bahmutov** [comment]
@akwasin your case seems pretty close to https://www.cypress.io/blog/2018/02/05/when-can-the-test-start/ - from the command log I see you visit the page and immediately do `cy.get(...).type(...)`. It seems you are typing before the page is ready to receive the input, maybe the page is hydrating or scripts are still loading. See if `cy.wait(1000)` can solve your problem. Of course the best solution is to wait for some external sign that the page is ready to receive the user's input, maybe a CSS class or similar attribute is set or removed when the app finishes bootstrapping.

**Comment by akwasin** [comment]
@bahmutov aha! that sounds reasonable. I got round the type issue by using .wait(500) as proposed by @davidyell but I'll test with a cy.wait(1000) after initially loading the page. 

@jennifer-shehane email incoming later today.

**Comment by mohsinht** [comment]
Facing same issue.

**Comment by guncebektas** [comment]
The issue still exists v.7.3.0
It's only happening on one input, i couldn't find the reason but adding .wait(500) solve the problem for me
```cy.get('input[name="customer-search-string"]').first().wait(500).clear().type(TEST.CUSTOMER.PHONE);```

**Comment by vazsonyidl** [comment]
Still exists in v.7.2.0 as well, but a workaround for this, you can use the following.

```
const httpSettingName = cy.get('[data-testid=cy-http-setting-name]'); // or you can select how you would like to
httpSettingName.wait(0).focus().clear().type('your_value_comes_here');
```

Waiting 0 ms means you do not slow down your tests but give Cypress enough time to "warm-up".

**Comment by akwasin** [comment]
> Still exists in v.7.2.0 as well, but a workaround for this, you can use the following.
> 
> ```
> const httpSettingName = cy.get('[data-testid=cy-http-setting-name]'); // or you can select how you would like to
> httpSettingName.wait(0).focus().clear().type('your_value_comes_here');
> ```
> 
> Waiting 0 ms means you do not slow down your tests but give Cypress enough time to "warm-up".

Interesting solution with 0 ms. Ill try that out

**Comment by sra1rdy** [comment]
This issue needs to be reopened as it still exists

**Comment by jennifer-shehane** [comment]
@sra1rdy If you can provide a reproducible example that we can run to see the issue, we'd be happy to reopen.

**Comment by bschley** [comment]
I had the same issue and I found a way to fix it.

The problems is sometimes when you try to type that Cypress lose the focus at the beginning of that elements for X ms (when its get xhr responses for example).

To prevent that focus loss at the beginning I did this:

`
cy.get('[data-cy="searchField"]')     
  **.click()
  .wait(500)**
  .type('thisisalooongteststriiiing{enter}')
`

**Comment by AbrahamBrookes** [comment]
Adding waits to each test or writing special event listeners into my application code just to tell the tests that it's ok to start testing really is not an acceptable solution here. I have gone with the solution provided by @jonnyparris - when this bug pops up, add a call to `clear()` and then re-`type()` into the input. This bug really shouldn't exist at all.

**Comment by akwasin** [comment]
@AbrahamBrookes Out of curiosity, do you reckon that its faster to type `.type('x').clear().type('text')` rather than `.wait(500).clear().type('text')` or .`wait(0).focus().clear().type('text') `written by @vazsonyidl that Ive personally started using. I think that double typing just to "warm up cypress" should be avoided.

**Comment by AbrahamBrookes** [comment]
Depends on how long it takes to type your warm up text I guess. My worry
with `wait` is that I just don't like how it smells. If the app takes
longer than the `wait` duration to boot for some reason then we're back
where we started. I feel like typing and clearing is at least interaction
with the app so maybe cypress will pick up a different error somehow.
Probably not though, this is just conjecture. I just don't like using lots
of waits.

**Comment by akwasin** [comment]
@AbrahamBrookes about your **edit**, You're right on your first point, I was referring to the speed of Cypress handling `.type('x').clear().type('text')` vs `.wait(0).focus().clear().type('text')` and not how fast we type on our keyboards.

**Comment by demetris-manikas** [comment]
Same problem here with 8.1.0.
It is probably a nasty timing issue since it appears half of the times I run the tests.

@jennifer-shehane The bug is certainly there. To provide an example demonstrating it is not easy since as I said it is 
not a bug that shows up all the times. Would a video be enough for you to admit it exists and reopen the issue? 
Thanks

**Comment by komali2** [comment]
We're having this issue. I'd love to make a reproducible example but it's likely that reproducing requires having a large app that has relatively large page load times, and our app isn't open source unfortunately. Anyway, seconding the solution others had, for us adding wait(1000) solved it, which we set to (2000) to be safe. Obviously annoying, but it got the job done.

**Comment by romankhomitskyi** [comment]
> We're having this issue. I'd love to make a reproducible example but it's likely that reproducing requires having a large app that has relatively large page load times, and our app isn't open source unfortunately. Anyway, seconding the solution others had, for us adding wait(1000) solved it, which we set to (2000) to be safe. Obviously annoying, but it got the job done.

Could you try plugin cypress real events to see if that works?
Use realClick, and realType

**Comment by KittyGarrison** [comment]
I am still having this issue as well
<img width="380" alt="Screen Shot 2022-02-14 at 3 09 32 PM" src="https://user-images.githubusercontent.com/11085480/153967920-3ace792b-f3d7-4727-9402-7430d6cbe072.png">

**Comment by demetris-manikas** [comment]
@akwasin Sorry to say but I fail see how waiting for half a second is a good suggestion.
@jennifer-shehane You have to admit that the problem exists and that no solution proposed here is a proper workaround. Please reopen the issue and so that the cypress developers or community find the bug and fix it.
Keeping this closed just does not make any sense.

**Comment by akwasin** [comment]
@demetris-manikas I totally agree but its currently one of the best functional workarounds to the current issue.
`.wait(0)`
also works if 500 ms is too much for you.

**Comment by demetris-manikas** [comment]
@akwasin Well... If `.click().wait(0)` works why doesn't cypress add it in the `.type()` code and save us all ?
I guess because it does not always work to only call `.wait(0)`.
Anyway...

**Comment by bahmutov** [comment]
Or you could not enable the input box until the app is you know, ready to process the input? Maybe have the disabled attribute and remove it only after the app has finished scaffolding?

**Comment by akwasin** [comment]
Or possibly add a setting to adjust execution speed, like playwrights slo-mo.

**Comment by khyamay** [comment]
I was having same issue and none of the workaround worked for me. I had to use `.type('{enter} some text'  )` and it seems to be working for me at least.

**Comment by othondev** [comment]
In my case, I found the issue with our app design that is causing it. We have a component that overlays all views when user data wasn't loaded that happens for a brief moment but enough to stop cypress typing. 

Example to reproduce it:
```
//test.spec.js
describe("Login and Logout flow", () => {
  const TIMES = 2;
  const something = "0123456789";
  for (let i = 0; i < TIMES; i++) {
    it(`Enter with god email and password ${i}`, () => {
      cy.visit("/");
      cy.get("#x").type(something);
      cy.get("#x").should("have.value", something);
    });
  }
});
```
```
//index.html
<!DOCTYPE html>
<html lang="en">
  <head>
    <link
      href="https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700,900"
      rel="stylesheet"
    />
    <link
      href="https://cdn.jsdelivr.net/npm/@mdi/font@6.x/css/materialdesignicons.min.css"
      rel="stylesheet"
    />
    <link
      href="https://cdn.jsdelivr.net/npm/vuetify@2.x/dist/vuetify.min.css"
      rel="stylesheet"
    />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no, minimal-ui"
    />
    <title>Clash</title>
    <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/vue@2.6.14/dist/vue.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/vuetify@2.x/dist/vuetify.js"></script>
    <script>
      let times = 30;
      document.addEventListener("DOMContentLoaded", function () {
        new Vue({
          el: "#app",
          vuetify: new Vuetify(),
          data: {
            hide: true,
            msg: "",
            dialog: false,
          },
          created: function () {
            setInterval(() => {
              if (--times > 0) {
                this.hide = !this.hide;
              } else {
                this.hide = false;
              }
              console.log({ times });
            }, 20);
          },
        });
      });
    </script>
  </head>
  <body>
    <div id="app">
      <v-app>
        <v-content>
          <v-layout>
            <p>Template 1</p>
            <input id="x" v-model="msg" />
            <p>{{ msg }}</p>
            <v-dialog v-model="hide" scrollable persistent max-width="950px">
              <v-card> Loading... </v-card>
            </v-dialog>
          </v-layout>
        </v-content>
      </v-app>
    </div>
  </body>
</html>
```
Test results:
![image](https://user-images.githubusercontent.com/28935584/155467721-687de991-bd0b-4bcf-8537-29a675537054.png)
Video:

https://user-images.githubusercontent.com/28935584/155467909-e46fbad6-51b0-4f12-9618-25ea88d81134.mp4

I hope this could help.

**Comment by romankhomitskyi** [comment]
The issue is still exists

**Comment by ra00f** [comment]
> Still exists in v.7.2.0 as well, but a workaround for this, you can use the following.
> 
> ```
> const httpSettingName = cy.get('[data-testid=cy-http-setting-name]'); // or you can select how you would like to
> httpSettingName.wait(0).focus().clear().type('your_value_comes_here');
> ```
> 
> Waiting 0 ms means you do not slow down your tests but give Cypress enough time to "warm-up".

Is this wait(0) documented somewhere?

**Comment by akwasin** [comment]
> Is this wait(0) documented somewhere?

Nope not that I know of and its extremely anti-pattern'ish but so the best working solution to missing letters.

https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting

**Comment by ry-sny** [comment]
I'm also experiencing the same issue. The only workaround that works is adding a hardcoded sleep but I'd like to avoid this if possible.

**Comment by maaaariana** [comment]
I'm also facing this issue((
@jennifer-shehane

**Comment by yuriy1715** [comment]
issue still exist!

**Comment by stychu** [comment]
Yeah, 10.4.0 and I still can experience this as well

**Comment by gpolanco** [comment]
If they haven't solved it after so long, it will be that the problem is more complicated than it seems (I hope), but still it's strange that nobody says anything about it

**Comment by romankhomitskyi** [comment]
> If they haven't solved it after so long, it will be that the problem is more complicated than it seems (I hope), but still it's strange that nobody says anything about it

Because there is no reproducible example, it can be to multiple reasons

**Comment by conversaShawn** [comment]
Right now there doesn't seem to be enough information to reproduce the problem on our end. We'll have to close this issue until we can reproduce it. This does not mean that your issue is not happening - it just means that we do not have a path to move forward.

Please open a new issue with a reproducible example and link to this issue. Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.
